prow/apis/prowjobs/v1/types.go

```diff
@@ -313,7 +313,13 @@ func (src *SlackReporterConfig) ApplyDefault(def *SlackReporterConfig) *SlackRep
 		merged.Host = def.Host
 	}
 	if merged.JobStatesToReport == nil {
-		merged.JobStatesToReport = def.JobStatesToReport
+		// check if src.JobStatesToReport is an empty slice, we want to keep it to
+		// silence alerts at the job level
+		if src.JobStatesToReport != nil {
```

Comment:
This code path is not covered by the new unit test added below

Also, is it possible that both `merged.JobStatesToReport == nil` and `src.JobStatesToReport != nil` are true? I think if the latter is true, then `merged` should be a deepcopy of `src`, and `JobStatesToReport` cannot be nil

Reply:
Thanks a lot for your review @chaodaiG thanks to it I see now that one of the existing test cases should cover most of the failing functionality, need to dig deeper.

Reply:
@chaodaiG I've verified that all is working as expected and #22533 is already fixed, sorry for the noise. Do you think it is worth adding a small note to crier's readme about how to silence slack notifications at the ProwJob level? I've updated this PR to do so.

Reply:
yeah, more doc is always better. Thank you!